### DISCUSSION TITLE
Add Symbol Hierarchy to the Outline View

### DISCUSSION THREAD

**Discussion by tehnrd** [forcedotcom/salesforcedx-vscode#5827] [Ideas]
Visual Studio code has released Sticky Scroll in version 1.70, which is awesome!
https://code.visualstudio.com/updates/v1_70#_editor-sticky-scroll 
https://www.tiktok.com/@vscode/video/7130009070798392618

But for this to work correctly, it requires the Outline section of VSCode to have a proper hierarchy of symbols. For some reason, the Outline for `.cls` files is completely flat, and this feature does not work with Apex code...

<img width="376" alt="Screen Shot 2022-08-11 at 9 30 05 PM" src="https://user-images.githubusercontent.com/1007085/184268836-1a3ec84f-786d-4be1-aec3-1bf2811eb942.png">

Adding proper hierarchy support in the Outline for Apex class files would enable this feature to work properly in VSCode.

Sorry if this is the wrong place to add this, point me to the correct repo if this is wrong, and I will create a request there.

**Comment by mshanemc** [comment]
seems to be working like you want now
<img width="452" height="434" alt="Screenshot 2026-04-22 at 10 06 57 PM" src="https://github.com/user-attachments/assets/b9a6cae1-47ea-4409-8d35-6db1461fbf3a" />
